### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by ddubeau**
Added separate subnets for PrivateLink VPC endpoint and replaced the AWS region drop down by an assertion rule

### CODE REVIEW

File: docs/partner_editable/deployment_options.adoc [hunk 1]
```diff
@@ -53,6 +53,86 @@ IMPORTANT: Choose the first option if you can create an IAM role in your account
             ],
             "Resource": "*"
         }
----- 
-
-
+----
```
Comment (sshvans) [inline docs/partner_editable/deployment_options.adoc:56]: Doc changes needs to be made as a separate PR to doc-edits branch. We are using separate branches for docs and code. 
Doc changes goes to doc-edits, and code changes goes to main.

**Comment by sshvans** [comment]
@ddubeau If you can remove the doc changes from this PR and submit it as a new PR to doc-edits branch, I'll be able to merge the code changes here.

File: docs/partner_editable/deployment_options.adoc (hunk 1 above)
Comment (ddubeau) [reply docs/partner_editable/deployment_options.adoc]: Thanks Shivansh. I've reverted the doc changes for now and I'll push a new PR in a few days. Could you please look into merging the code changes ASAP?

**Review by sshvans — APPROVED** [review]
